torch/fx/graph_module.py

```diff
@@ -276,6 +264,113 @@ def _gen_model_repr(module_name: str, module: torch.nn.Module) -> Optional[str]:
             warnings.warn("Was not able to save the following children modules as reprs -"
                           f"saved as pickled files instead: {blobified_modules}")
 
+    def has_submodule(self, target: str) -> bool:
+        """
+        Returns whether or not this GraphModule contains the submodule
+        given by ``str``.
+
+        Args:
+            target: The fully-qualified string name of the submodule
+                to look for.
+        """
+        atoms: List[str] = target.split(".")
+        mod: torch.nn.Module = self
+
+        for item in atoms:
+            if not hasattr(mod, item):
+                return False
+            mod = getattr(mod, item)
```

Comment:
`getattr(mod, item)` isn't guaranteed to always return an `nn.Module`. Probably want to have an `isinstance` check here and return false if that fails. By definition of this method, I think this should return `false` if passed `foo.bar.baz` and `foo.bar.baz` is a parameter/other attribute but not a module